### PR TITLE
Use crane for Nix builds

### PR DESCRIPTION
**Pull request by 9999years**
[Crane](https://crane.dev/) is a Nix framework for building Rust packages. The main benefit is that it separates the build of the dependency packages into a separate derivation, so that those artifacts can be reused between builds and CI runs.

### CODE REVIEW

**Comment by 9999years** [comment]
Current dependencies on/for this PR:
* main
  * **PR #54** <a href="https://app.graphite.dev/github/pr/MercuryTechnologies/ghcid-ng/54" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/MercuryTechnologies/ghcid-ng/54?utm_source=stack-comment).

File: src/fake_reader.rs [hunk 1]
```diff
@@ -18,13 +18,6 @@ pub struct FakeReader {
 }
 
 impl FakeReader {
-    /// Construct a `FakeReader` from an iterator of bytes.
```
Comment (9999years) [inline src/fake_reader.rs:21]: These methods were never used.

File: src/ghci/mod.rs [hunk 2]
```diff
@@ -126,12 +126,11 @@ impl Ghci {
...
-              GhciStdin {
-                stdin,
-                stdout_sender: stdout_sender.clone(),
-                stderr_sender: stderr_sender.clone(),
-              };
+        let stdin = GhciStdin {
+            stdin,
+            stdout_sender: stdout_sender.clone(),
+            stderr_sender: stderr_sender.clone(),
+        };
```
Comment (9999years) [inline src/ghci/mod.rs:133]: The formatter wanted this change.

File: .github/workflows/version.yaml [hunk 3]
```diff
@@ -105,16 +105,6 @@ jobs:
         run: |
           echo "version=$(./.github/workflows/get-crate-version.sh)" >> "$GITHUB_OUTPUT"
 
-      - name: Increment `flake.nix` version
```
Comment (9999years) [inline .github/workflows/version.yaml:108]: The `flake.nix` no longer contains a version number.

File: flake.nix [hunk 4]
```diff
@@ -3,6 +3,14 @@
 
   inputs = {
     nixpkgs.url = "github:NixOS/nixpkgs";
+    crane = {
+      url = "github:ipetkov/crane";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
```
Comment (evanrelf) [inline flake.nix:9]:
```suggestion
    crane = {
      url = "github:ipetkov/crane";
      inputs.flake-compat.follows = "flake-compat";
      inputs.flake-utils.follows = "flake-utils";
      inputs.nixpkgs.follows = "nixpkgs";
    };
```

**Review by evanrelf — APPROVED** [review]

**Review by evanrelf — DISMISSED** [review]

File: flake.nix [hunk 5]
```diff
@@ -13,68 +23,84 @@
...
+        ghcid-ng = craneLib.buildPackage (commonArgs
+          // {
+            # Don't run tests; we'll do that in a separate derivation.
+            # This will allow people to install and depend on `ghcid-ng`
+            # without downloading a half dozen different versions of GHC.
+            doCheck = false;
+          });
+      in {
+        checks = {
+          inherit ghcid-ng;
```
Comment (Gabriella439) [inline flake.nix:75]: do you need the test-free `ghcid-ng` underneath `checks`?

**Review by Gabriella439 — APPROVED** [review]

File: flake.nix (hunk 5 above)
Comment (9999years) [reply flake.nix:75]: Oh, good catch.

**Review by Gabriella439 — APPROVED** [review]

**Pull request merged**